library/src/scripts/components/Plural.tsx

```diff
@@ -0,0 +1,32 @@
+/**
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import * as React from "react";
+import classNames from "classnames";
+import Translate from "@library/components/translation/Translate";
+
+interface IProps {
+    value: number;
+    singular?: string;
```

Comment:
I feel like you have this backwards. `plural` should be the optional prop and singular the required, because that's what you checking in the render method.